File: backend/src/main/java/com/isravel/workhub/attendance/AttendanceLog.java
package com.isravel.workhub.attendance;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "attendance_logs", uniqueConstraints = @UniqueConstraint(columnNames = {"employee_id", "punch_time"}))
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AttendanceLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employee_id", nullable = false)
    private Long employeeId;

    @Column(name = "punch_time", nullable = false)
    private LocalDateTime punchTime;

    @Column(name = "verify_mode")
    private Integer verifyMode;

    @Column(name = "status")
    private Integer status;

    @Column(name = "created_at")
    private LocalDateTime createdAt;
}

File: flutter_app/lib/models/attendance_log.dart
class AttendanceLog {
  final int id;
  final int employeeId;
  final String punchTime;
  final String? verifyMode;
  final String? status;
  final String? createdAt;

  AttendanceLog({
    required this.id,
    required this.employeeId,
    required this.punchTime,
    this.verifyMode,
    this.status,
    this.createdAt,
  });

  factory AttendanceLog.fromJson(Map<String, dynamic> json) {
    return AttendanceLog(
      id: json['id'],
      employeeId: json['employeeId'],
      punchTime: json['punchTime'],
      verifyMode: json['verifyMode']?.toString(),
      status: json['status']?.toString(),
      createdAt: json['createdAt'],
    );
  }
}
